armi/reactor/blueprints/assemblyBlueprint.py

```diff
@@ -175,6 +175,9 @@ def _constructAssembly(self, cs, blueprint):
         aziMeshPoints = self.azimuthalMeshPoints or 1
         a.p.AziMesh = aziMeshPoints
 
+        # TODO: How can this be done better?
+        a.p.THcoolantInletT = cs["Tin"]
```

Comment:
It might be nice to have an assembly method in `assemblies.py` that is something like:

```
def inletTemperatureSetter(self, temperatureInC):
    a.p.THcoolantInletT = temperatureInC
```

In this way we can reuse the temperature setting functionality without directly setting the parameter.

Reply:
Discussed offline some and addressed here
 https://github.com/terrapower/armi/pull/993/commits/abfaede115874ce52ead2cf2f222640f0d5d4d12

Reply:
The implication of setting this should be very clear. For example, what do we do if the grid plate dimensions and the pitch of the assembly defined in the blueprints is not in sync with this temperature assignment? We now have two ways of setting up the reactor, one mechanism being that we do so with all cold dimensions and the other being setting up with hot dimensions for the blocks. The pitch of the assembly has typically been setup to assume that we are talking about hot dimensions (i.e., already expanded) at the inlet temperature (`cs["Tin"]`), but if this isn't the case then does this assignment made sense?

Reply:
> what do we do if the grid plate dimensions and the pitch of the assembly defined in the blueprints is not in sync with this temperature assignment?

By default, the grid plate dimensions and temperature of the grid plate should always be in sync. But as you point out, there is a chance that the grid plate dims/temp could be out of sync with the assembly inlet temperature and pitch, `cs["Tin"]` and `core.spatialGrid._unitSteps`, respectively. This seems problematic to me and an excellent point to bring up - not to mention unphysical (the grid plate and inlet temperature should probably be in sync). We should likely add checks to ensure alignment between the grid plate temp and assembly inlet temp at BOL and whenever the grid plate pitch is updated.

Reply:
the intercoolant outer pitch is the dimension that determines the lattice.

It is defined at the hot temperature. even if the intercoolant changes temperature the OP should not change dimension even if the intercoolant temp changes. it should only change dimension if the grid plate temp changes. 

there is no way to check if they are consistent because the cold grid plate lattice is not defined anywhere.

Reply:
I think it's reasonable that the user should keep these consistent on initialization. Maybe just a runLog would help us keep a check on this. One other suggestion could be to define the cold grid plate pitch as a setting that could then be verified against the temperature assignment of Tin and the defined assembly pitch. 

I am fine with it being a an expectation that the user defines this correctly but some communication to the user in the stdout that says "hey, I'm initializing the core pitch to be X with an associated temperature of Y". This is something that we could check. Also, one thing that might help is a lot is to add a core parameter for pitch. It's actually not clear to me if loading a snapshot case after grid plate dimension changes will stick. I think they would because this would be reflected in dimension changes to components but I am not certain that I understand how this works.

Sorry for the non-answer, trying to think through it aloud and get more feedback before we decide what makes the most sense